client_plugin/drivers/shared/shared_driver.go

```diff
@@ -62,6 +86,18 @@ func NewVolumeDriver(cfg config.Config, mountDir string) *VolumeDriver {
 	}
 	d.dockerd = cli
 
+	// Pull Samba image
+	log.Infof("Pulling Samba server image")
```

Comment:
Your guessing is  correct 

 > But downloading during plugin creation would mean that the samba server will become a part of the plugin.

No, it will mean that samba container will be pre-loaded into Docker cache, and we will know for sure it will fit